cognite/client/utils/_time.py

```diff
@@ -484,15 +484,13 @@ def _to_fixed_utc_intervals_fixed_unit_length(
     utc = get_utc_zoneinfo()
 
     freq = multiplier * GRANULARITY_IN_HOURS[unit]
-    index = pandas_date_range_tz(start, end, f"{freq}H")
-    expected_freq = pd.Timedelta(hours=freq)
-    next_mask = (index - index.to_series().shift(1)) != expected_freq
-    last_mask = (index.to_series().shift(-1) - index) != expected_freq
-    index = index[last_mask | next_mask]
+    index = pandas_date_range_tz(start, end, to_pandas_freq(f"{multiplier}{unit}", start))
+    utc_offsets = index.to_series().apply(lambda t: t.utcoffset())
+    transition_raw = index[(utc_offsets != utc_offsets.shift(-1)) | (utc_offsets != utc_offsets.shift(1))]
```

Comment:
Last thought: we now have every single UTC offset in the time range; should we raise if not all are whole-hour offsets from UTC? Seems like a great opportunity 🤩

Reply:
That is a great point, only thing is that it would actually not be raised if the granularity is variable (1 month or more) as this code is run for fixed cases. I will try to do some refactoring and then add the check, so it is raised in all circumstances.

Reply:
Then I suggest leaving it for a future PR 👍

Reply:
Good, will add it to my list